### PR TITLE
Settings page refactoring

### PR DESCRIPTION
**Pull request by okhmelevskyi**
**Created new files and folders:**

- ReportsFolderSection;
- ThemeSection;
- ConnectionsSection;
- VersionSection.

### CODE REVIEW

File: renderer/pages/settings.tsx [hunk 1]
```diff
@@ -1,84 +1,63 @@
...
-    shallow
+    shallow,
   );
   const [isOSDarkTheme, setIsOSDarkTheme] = useState(true);
   const [currentMenuItem, setCurrentMenuItem] = useState<SidebarNavItem>(
-    SidebarNavItem.Connections
+    SidebarNavItem.Connections,
   );
 
+  const settingSections = SETTING_SECTIONS[currentMenuItem];
```
Comment (a-nahaitsev) [inline renderer/pages/settings.tsx:20]: I guess this variable should be in singular, like `settingSection` or `currentSettingSection` or `activeSettingSection`

File: renderer/pages/settings.tsx [hunk 2]
```diff
@@ -1,84 +1,63 @@
...
-      case SidebarNavItem.Theme:
-        return <ThemeSection />;
-
-      case SidebarNavItem.VersionSelect:
-        return <VersionSection />;
-    }
-  };
+  const renderSidebarNavItem = () =>
+    Object.values(SidebarNavItem).map((value) => (
+      <li
```
Comment (a-nahaitsev) [inline renderer/pages/settings.tsx:42]: and, if I'm not mistaken, this variable should be in plural, like `renderSidebarNavItems`

**Review by a-nahaitsev — CHANGES_REQUESTED** [review]

File: renderer/pages/settings.tsx (hunk 1 above)
Comment (okhmelevskyi) [reply renderer/pages/settings.tsx:20]: Sure

File: renderer/pages/settings.tsx (hunk 2 above)
Comment (okhmelevskyi) [reply renderer/pages/settings.tsx:42]: Yes, you're right.

File: renderer/pages/settings.tsx (hunk 1 above)
Comment (okhmelevskyi) [reply renderer/pages/settings.tsx]: Resolved.

File: renderer/pages/settings.tsx (hunk 2 above)
Comment (okhmelevskyi) [reply renderer/pages/settings.tsx]: Resolved.

**Review by a-nahaitsev — APPROVED** [review]

**Review by o-mehlii — APPROVED** [review]

**Pull request merged**